### PR TITLE
Types for Header component

### LINKED ISSUE #564: Add types for Header component

### PR DESCRIPTION
**Pull request by marcialfps**
Closes #564 

## Changes

- Props types.
- Fix type errors:
   - `isResponsive` default value to false.
   - `ContentContainerComponent` receives children parameter.
- Documentation:
   - `content` prop type changed to `node`.

### CODE REVIEW

**Comment by GomezIvann** [comment]
Should we remove _Icons.jsx_ and add the icons inside the component, as it is done in Select or TextInput? Just to remove the .jsx file.

**Review by GomezIvann — CHANGES_REQUESTED** [review]

File: lib/src/header/Header.tsx [hunk 1]
```diff
@@ -52,7 +60,7 @@ const DxcHeader = ({ underlined = false, onClick, content, responsiveContent, ma
       }
     }
   };
-  const ContentContainerComponent = () => {
+  const ContentContainerComponent = (props: any) => {
```
[the remark this answers is not in the document]
Comment (marcialfps) [reply lib/src/header/Header.tsx:63]: If I remove the parameter the following type error will be displayed:
image

[the remark this answers is not in the document]
Comment (GomezIvann) [reply lib/src/header/Header.tsx:76]: Why does `ContentContainerComponent` have `responsiveContent(handleMenu)` both as a children and inside the function component? Children are not been used in the function component so maybe removing them is not a bad idea.

[the remark this answers is not in the document]
Comment (marcialfps) [reply lib/src/header/Header.tsx]: Yes! I didn't realize. :expressionless: I changed it in the last commit.

**Review by GomezIvann — APPROVED** [review]

**Pull request merged**